src/01/depository.ts

```diff
@@ -0,0 +1,183 @@
+import { BN, Program } from '@project-serum/anchor';
+import {
+  Cluster,
+  ConfirmOptions,
+  Connection,
+  PublicKey,
+} from '@solana/web3.js';
+import {
+  Num,
+  ZERO_ONE_DEVNET_PROGRAM_ID,
+  ZERO_ONE_MAINNET_PROGRAM_ID,
+  ZO_DEVNET_STATE_KEY,
+  ZO_FUTURE_TAKER_FEE,
+  ZO_MAINNET_STATE_KEY,
+} from '@zero_one/client';
+import { findAddrSync } from '../utils';
+import { IDL } from '../idl';
+import { ZoDepositoryAccount } from '../interfaces';
+import { Zo } from '.';
+
+const SLIPPAGE_BASIS = 1000;
+
+export class ZoDepository {
+  public pda: PublicKey;
+  public zoAccountPda: PublicKey;
+  public collateralMint: PublicKey;
+  public collateralMintSymbol: string;
+  public collateralMintDecimals: number;
+  public quoteMint: PublicKey;
+  public quoteMintSymbol: string;
+  public quoteMintDecimals: number;
+  uxdProgramId: PublicKey;
+
+  public constructor(
+    mint: PublicKey,
+    mintName: string,
+    mintDecimals: number,
+    quoteMint: PublicKey,
+    quoteMintName: string,
+    quoteMintDecimals: number,
+    uxdProgramId: PublicKey,
+    cluster: Cluster
+  ) {
+    let zoProgramId: PublicKey;
+    let zoState: PublicKey;
+    switch (cluster) {
+      case 'devnet':
+        zoProgramId = ZERO_ONE_DEVNET_PROGRAM_ID;
+        zoState = ZO_DEVNET_STATE_KEY;
+        break;
+      case 'mainnet-beta':
+        zoProgramId = ZERO_ONE_MAINNET_PROGRAM_ID;
+        zoState = ZO_MAINNET_STATE_KEY;
+        break;
+      default: {
+        throw new Error(`Invalid ZeroOne cluster: ${cluster}`);
+      }
+    }
+    this.collateralMint = mint;
+    this.collateralMintSymbol = mintName;
+    this.collateralMintDecimals = mintDecimals;
+    this.quoteMint = quoteMint;
+    this.quoteMintSymbol = quoteMintName;
+    this.quoteMintDecimals = quoteMintDecimals;
+    this.pda = findAddrSync(
+      [Buffer.from('ZODEPOSITORY'), mint.toBuffer()],
```

Comment:
`Buffer` is a Node.js API, it's not available in the browser, which means consumers who'll want to use `uxd-client` for an app, will need to tweak their bundler configuration to include a browser-compatible `Buffer` module & make it available for global use.

I didn't see we did that as well in other places, it's fine for now but better not to do this & include a buffer module dependency instead at some point.

Reply:
Leaving as-is for now, may fix in another PR